### ISSUE TITLE
Failed to build from source with cgal 6.0

### ISSUE THREAD

**Issue by sukanka** [IfcOpenShell/IfcOpenShell#5557]
Version: `0.8.1-alpha241010`
system: Archlinux x86_64 Linux 6.11.2-zen1-1-zen

I use [this PKGBUILD](https://gist.github.com/sukanka/b340e30b4457fb1a46015d7746e973c0) to build the package, but failed with error messages: (full log  [ifcopenshell-0.8.1_alpha241010-1-x86_64-build.log](https://github.com/user-attachments/files/17328304/ifcopenshell-0.8.1_alpha241010-1-x86_64-build.log) )

```
/build/ifcopenshell/src/IfcOpenShell-bonsai-0.8.1-alpha241010/cmake/../src/ifcconvert/validation_utils.h:226: error: type/value mismatch at argument 1 in template parameter list for ‘template<long unsigned int Idx, class T, long unsigned int N> T& boost::get(array<T, N>&)’
  226 |                                 if (boost::get<Point>(&(intersection->first))) {
/build/ifcopenshell/src/IfcOpenShell-bonsai-0.8.1-alpha241010/cmake/../src/ifcconvert/validation_utils.h:226: note:   expected a constant of type ‘long unsigned int’, got ‘remove_thickness::Point’ {aka ‘CGAL::Point_3<CGAL::Epeck>’}
/usr/include/boost/array.hpp:428:13: note: candidate: ‘template<long unsigned int Idx, class T, long unsigned int N> const T& boost::get(const array<T, N>&)’
  428 |    const T &get(const boost::array<T,N> &arr) BOOST_NOEXCEPT {
      |             ^~~
```
I guess it has something to do with cgal 6.0  instead of boost 1.86, as archlinux has upgraded boost to 1.86 on  2024.08.29 and I built `ifcopenshell-0.8.1-alpha20240902` successfully on 2024.09.02. 

Thanks.

**Comment by aothms** [comment]
Yes, it's likely a boost thing, because the function prototype being referenced is:

```
boost::get(array<T, N>&)
```

From the cgal docs it's very obtuse what the type is supposed to be, but it's likely not an array, but more likely a variant. But the boost::get for array requires an integer and not a type as the first template arg and hence the message:

```
error: type/value mismatch at argument 1 
```

This is low quality code anyway that should just be removed, but in the meantime I would find the quickest way to solve this error by commenting it out or wrapping it in a `if constexpr (false)` or sth.

**Comment by sukanka** [comment]
> This is low quality code anyway that should just be removed, but in the meantime I would find the quickest way to solve this error by commenting it out or wrapping it in a `if constexpr (false)` or sth.

Thanks, commenting these codes out works for me.

**Comment by sukanka** [comment]
resolved in the latest release.
